props/saber_chris_buttons.h

```diff
@@ -0,0 +1,1036 @@
+
+Tightened click timings
+I've shortened the timeout for short and double click detection from 500ms
+to 300ms.  I think it feels more responsive this way but still gives enough
+timeout to ensure all button actions can be achieved consistently
+I've included all button timings so they can be easily tweaked to suit
+individual tastes. MM.
+
+============================================================
+============================================================
+*/
+#ifndef PROPS_SABER_SENSE_BUTTONS_H
+#define PROPS_SABER_SENSE_BUTTONS_H
+
+
+//  Used instead of BladeID and allows toggle switching 
+//  between two blade/preset arrays manually.
+#ifdef FAKE_BLADE_ID
+struct FakeBladeID {
+   static int return_value;
+   float id() { return return_value; }
+   static void toggle() {
+     return_value = NO_BLADE - return_value;
+   }
+};
+int FakeBladeID::return_value = 0;
+#undef BLADE_ID_CLASS_INTERNAL
+#define BLADE_ID_CLASS_INTERNAL FakeBladeID
+#undef BLADE_ID_CLASS
+#define BLADE_ID_CLASS FakeBladeID
+#endif
+
+
+#include "prop_base.h"
+#include "../sound/hybrid_font.h"
+
+#undef PROP_TYPE
+#define PROP_TYPE SabersensePropV31
+
+#ifndef MOTION_TIMEOUT
+#define MOTION_TIMEOUT 60 * 15 * 1000
+#endif
+
+#ifndef SA22C_SWING_ON_SPEED
+#define SA22C_SWING_ON_SPEED 250
+#endif
+
+#ifndef SA22C_LOCKUP_DELAY
+#define SA22C_LOCKUP_DELAY 200
+#endif
+
+#ifndef SA22C_FORCE_PUSH_LENGTH
+#define SA22C_FORCE_PUSH_LENGTH 5
+#endif
+
+#ifndef BUTTON_DOUBLE_CLICK_TIMEOUT
+#define BUTTON_DOUBLE_CLICK_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_SHORT_CLICK_TIMEOUT
+#define BUTTON_SHORT_CLICK_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_HELD_TIMEOUT
+#define BUTTON_HELD_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_HELD_MEDIUM_TIMEOUT
+#define BUTTON_HELD_MEDIUM_TIMEOUT 1000
+#endif
+
+#ifndef BUTTON_HELD_LONG_TIMEOUT
+#define BUTTON_HELD_LONG_TIMEOUT 2000
+#endif
+
+#ifdef SA22C_SWING_ON
+#define SWING_GESTURE
+#endif
+
+#ifdef SA22C_STAB_ON
+#define STAB_GESTURE
+#endif
+
+#ifdef SA22C_TWIST_ON
+#define TWIST_GESTURE
+#endif
+
+#ifdef SA22C_THRUST_ON
+#define THRUST_GESTURE
+#endif
+
+
+//  #define FORCE_PUSH_CONDITION battle_mode_
+
+EFFECT(dim);      // for EFFECT_POWERSAVE
+EFFECT(battery);  // for EFFECT_BATTERY_LEVEL
+EFFECT(bmbegin);  // for Begin Battle Mode
+EFFECT(bmend);    // for End Battle Mode
+EFFECT(vmbegin);  // for Begin Volume Menu
+EFFECT(vmend);    // for End Volume Menu
+EFFECT(volup);    // for increse volume
+EFFECT(voldown);  // for decrease volume
+EFFECT(volmin);   // for minimum volume reached
+EFFECT(volmax);   // for maximum volume reached
+EFFECT(faston);   // for EFFECT_FAST_ON
+EFFECT(blstbgn);  // for Begin Multi-Blast
+EFFECT(blstend);  // for End Multi-Blast
+EFFECT(push);     // for Force Push gesture in Battle Mode
+EFFECT(quote);    // for playing quotes
+
+
+
+// The Saber class implements the basic states and actions
+// for the saber.
+class SabersensePropV31 : public PROP_INHERIT_PREFIX PropBase {
```

Comment:
Please use consistent naming, either "chris" or "sabersense".

The class name, the file and the defines should all use the same thing.

Reply:
Also, putting a version in the name is interesting. However, if you do that I won't let you just rename it when you make changes, you will need to have a backwards-compatible "SaberSensePropV31", so that older configurations don't stop working all of a sudden.

Reply:
Understood on the version naming. The idea was that assuming things get tweaked or updated later on, that typing 'version' in Serial Monitor would show which edition of the prop had been uploaded to a given saber. But your point about backwards compatibility is noted and wholeheartedly agreed with.
And while we're on the subject, on a brief related side-note, if I've understood it right, this pull request is against the GitHub Master - do I need to do a separate pull request against ProffieOS 7.x?

Reply:
This prop will not be a part of ProffieOS 7. It will be a part of ProffieOS 8.